src/game/scenes/mainmenu/menu_video.c

```diff
@@ -80,6 +80,11 @@ void resolution_toggled(component *c, void *userdata, int pos) {
     }
 }
 
+void renderer_toggled(component *c, void *userdata, int pos) {
+    settings_video *v = &settings_get()->video;
+    video_get_renderer_info(pos, &v->renderer, NULL);
```

Comment:
settings' F_STRING is for owned strings, not string constants (this currently crashes on exit if the settings have been opened).
v->renderer should be a `char*`, and renderer_toggled needs to free and allocate (or do both at once with realloc).

Reply:
https://github.com/omf2097/openomf/pull/774/files